apis/object/v1alpha1/types.go

```diff
@@ -18,10 +18,61 @@ package v1alpha1
 
 import (
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// A ManagementPolicy determines what should happen to the underlying external
+// resource when a managed resource is created, updated, deleted, or observed.
+// +kubebuilder:validation:Enum=Default;ObserveCreateUpdate;ObserveDelete;Observe
+type ManagementPolicy string
+
+const (
+	// Default means the provider can fully manage the resource.
+	Default ManagementPolicy = "Default"
+	// ObserveCreateUpdate means the provider can observe, create, or update
+	// the resource, but can not delete it.
+	ObserveCreateUpdate ManagementPolicy = "ObserveCreateUpdate"
+	// ObserveDelete means the provider can observe or delete the resource, but
+	// can not create and update it.
+	ObserveDelete ManagementPolicy = "ObserveDelete"
+	// Observe means the provider can only observe the resource.
+	Observe ManagementPolicy = "Observe"
+)
+
+// FromObject refers to an object by Name, Kind, APIVersion, etc. It is used
+// to reference other Object or arbitrary Kubernetes resource which is either
+// cluster or namespace scoped.
+type FromObject struct {
+	// APIVersion of the referenced object.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referenced object.
+	Kind string `json:"kind"`
+	// Name of the referenced object.
+	Name string `json:"name"`
```

Comment:
I believe at some point we will need to add support for label selectors similar to existing crossplane references with selectors, e.g. `vpcIdSelector`, but it is totally ok (even preferable) to postpone it to later PR.

Reply:
Totally agree! We can do it iteratively.